src/main/scala/com/databricks/spark/xml/XmlReader.scala

```diff
@@ -119,14 +119,11 @@ class XmlReader extends Serializable {
 
   /**
    * @param spark current SparkSession
-   * @param df XML for individual 'rows' as Strings
+   * @param ds XML for individual 'rows' as Strings
    * @return XML parsed as a DataFrame
    */
-  def xmlDataFrame(spark: SparkSession, df: DataFrame): DataFrame = {
-    require(df.columns.length == 1 && df.schema.fields.head.dataType == StringType,
-      "DataFrame must have a single String column containing XML")
-    import spark.implicits._
-    xmlRdd(spark, df.as[String].rdd)
+  def xmlDataset(spark: SparkSession, ds: Dataset[String]): DataFrame = {
```

Comment:
WDYT @srowen? I think it's just fine to have one `xmlDataset` since they can be easily converted to each other.

Reply:
Sure OK, that's fine too